src/multisig/multisigController.ts

```diff
@@ -0,0 +1,833 @@
+import BigNumber from "bignumber.js";
+import { Abi } from "../abi";
+import {
+    Address,
+    BaseControllerInput,
+    IAccount,
+    Transaction,
+    TransactionsFactoryConfig,
+    TransactionWatcher,
+} from "../core";
+import { INetworkProvider } from "../networkProviders/interface";
+import { SmartContractController } from "../smartContracts";
+import { MultisigTransactionsFactory } from "./multisigTransactionsFactory";
+import { MultisigTransactionsOutcomeParser } from "./multisigTransactionsOutcomeParser";
+import * as resources from "./resources";
+
+export class MultisigController extends SmartContractController {
+    private transactionAwaiter: TransactionWatcher;
+    private multisigFactory: MultisigTransactionsFactory;
+    private multisigParser: MultisigTransactionsOutcomeParser;
+
+    constructor(options: { chainID: string; networkProvider: INetworkProvider; abi: Abi }) {
```

Comment:
we should also have an optional addressHrp parameter, that can be used instead of altering the LibraryConfig.

Reply:
To be discussed, if decided so will do in a different pr